parser.y

```diff
@@ -2341,6 +2341,13 @@ BeginTransactionStmt:
 			Mode: ast.Optimistic,
 		}
 	}
+|	"BEGIN" "ISOLATION" "LEVEL" IsolationLevel
+	// supports for setting isolation level when starting transaction,
+	// currently it doesn't actually set anything
+	// TODO: Set isolation level before starting transaction
+	{
+		$$ = &ast.BeginStmt{}
+	}
```

Comment:
You should use a parameter in begin stmt to save isolation level.

Reply:
It should now capture the correct isolation level